core/src/banking_stage/consumer.rs

```diff
@@ -684,6 +709,39 @@ impl Consumer {
         }
     }
 
+    pub fn check_fee_payer_unlocked(
+        bank: &Bank,
+        message: &SanitizedMessage,
+        error_counters: &mut TransactionErrorMetrics,
+    ) -> Result<(), TransactionError> {
+        let fee_payer = message.fee_payer();
+        let budget_limits =
+            process_compute_budget_instructions(message.program_instructions_iter())?.into();
+        let fee = bank.fee_structure.calculate_fee(
+            message,
+            bank.get_lamports_per_signature(),
```

Comment:
I noted early it should fetch `lamports_per_signature` for durable transactions from nonce account. Then @t-nelson just clarified that durable transaction's fee is calculated based on fee rate at execution. So this should be just fine.